src/Types/DateImmutableType.php

```diff
@@ -31,7 +26,7 @@ public function convertToDatabaseValue(mixed $value, AbstractPlatform $platform)
 
         throw InvalidType::new(
             $value,
-            $this->getName(),
+            $this::class,
```

Comment:
Although the end result would be the same and it's not enforced by the coding standard, `static::class` would make more sense here: the error comes from the type class itself, not from a given type instance (`$this`).